e2e/fiat_tf_test.go

```diff
@@ -0,0 +1,323 @@
+	require.NoError(t, err)
+	require.Zero(t, res.Code)
+
+	// ACTION: Grant an authz SEND using a TF token from a granter who is blacklisted
+	// EXPECTED: Success;
+
+	w = interchaintest.GetAndFundTestUsers(t, ctx, "default", math.OneInt(), noble, noble)
+	granter3 := w[0]
+	grantee3 := w[1]
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, granter3)
+
+	res, err = val.AuthzGrant(ctx, granter3, grantee3.FormattedAddress(), "send", "--spend-limit=100uusdc")
+	require.NoError(t, err)
+	require.Zero(t, res.Code)
+}
+
+func TestFiatTFAuthzSend(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+	t.Parallel()
+
+	ctx := context.Background()
+
+	nw := nobleSpinUp(t, ctx, true)
+	noble := nw.chain
+	val := noble.Validators[0]
+
+	// setup
+	w := interchaintest.GetAndFundTestUsers(t, ctx, "default", math.OneInt(), noble, noble, noble)
+	granter := w[0]
+	grantee := w[1]
+	receiver := w[2]
+
+	mintAmount := int64(100)
+	_, err := val.ExecTx(ctx, nw.fiatTfRoles.Minter.KeyName(), "fiat-tokenfactory", "mint", granter.FormattedAddress(), fmt.Sprintf("%duusdc", mintAmount))
+	require.NoError(t, err, "error minting")
+
+	res, err := val.AuthzGrant(ctx, granter, grantee.FormattedAddress(), "send", "--spend-limit=100uusdc")
+	require.NoError(t, err)
+	require.Zero(t, res.Code)
+
+	sendAmount := 5
+	nestedCmd := []string{
+		noble.Config().Bin,
+		"tx", "bank", "send", granter.FormattedAddress(), receiver.FormattedAddress(), fmt.Sprintf("%duusdc", sendAmount),
+		"--from", granter.FormattedAddress(), "--generate-only",
+		"--chain-id", noble.GetNode().Chain.Config().ChainID,
+		"--node", noble.GetNode().Chain.GetRPCAddress(),
+		"--home", noble.GetNode().HomeDir(),
+		"--keyring-backend", keyring.BackendTest,
+		"--output", "json",
+		"--yes",
+	}
+
+	// ACTION: Execute an authz SEND using a TF token from a grantee who is blacklisted
+	// EXPECTED: Request fails; Even though grantee is acting on behalf of the granter,
+	// the granter still cannot execute `send` due to being blacklisted
+	// Status:
+	// 	Granter1 has authorized Grantee1 to send 100usdc from their wallet
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, grantee)
+
+	_, err = val.AuthzExec(ctx, grantee, nestedCmd)
+	require.ErrorContains(t, err, fmt.Sprintf("an address (%s) is blacklisted and can not receive tokens: unauthorized", grantee.FormattedAddress()))
+
+	bal, err := noble.GetBalance(ctx, receiver.FormattedAddress(), "uusdc")
+	require.NoError(t, err)
+	require.True(t, bal.IsZero())
+
+	unblacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, grantee)
+
+	// ACTION: Execute an authz SEND using a TF token from a granter who is blacklisted
+	// EXPECTED: Request fails; Granter is blacklisted
+	// Status:
+	// 	Granter1 has authorized Grantee1 to send 100usdc from their wallet
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, granter)
+
+	preSendBal := bal
```

Comment:
Given that balance stays zero for this entire test (all requests fail), do we need this `preSendBal` variable? Can we always check that it is zero instead? Or is there a reason you prefer to do the checks in this way?

Reply:
Yes, good call. Simplified: 
[`133af8f` (#373)](https://github.com/noble-assets/noble/pull/373/commits/133af8f19831312ba7545a8a7270db1740c66e5e)